lib/components/Modal/Bottom/index.tsx

```diff
@@ -0,0 +1,160 @@
+import ReactDOM from "react-dom";
+import { useState, useEffect, useRef } from "react";
+import clsx from "clsx";
+import "./modal-bottom.scss";
+import { useSwipeable } from "react-swipeable";
+import { ModalImage } from "./modal-image";
+import { ModalTitle } from "./modal-title";
+import { ModalBody } from "./modal-body";
+import { Button } from "../../Button";
+
+interface ModalBottomProps {
+    isOpened?: boolean;
+    className?: string;
+    showHandleBar?: boolean;
+    showSecondaryButton?: boolean;
+    shouldCloseOnPrimaryButtonClick?: boolean;
+    toggleModal: (isOpened: boolean) => void;
+    portalId?: string;
+    primaryButtonLabel?: React.ReactNode;
+    primaryButtonFunction?: () => void;
+    secondaryButtonLabel?: React.ReactNode;
+}
+
+const swipeConfig = {
+    delta: 10,
+    trackTouch: true,
+    trackMouse: true,
+    preventScrollOnSwipe: true,
+};
+
+const maxExpandedHeight = 85;
+
+export const ModalBottom = ({
+    isOpened = false,
+    className,
+    children,
+    showHandleBar = true,
+    showSecondaryButton = true,
+    shouldCloseOnPrimaryButtonClick = false,
+    toggleModal,
+    portalId,
+    primaryButtonLabel,
+    primaryButtonFunction,
+    secondaryButtonLabel,
+}: React.PropsWithChildren<ModalBottomProps>) => {
+    const [isVisible, setIsVisible] = useState(isOpened);
+    const [isExpanded, setIsExpanded] = useState(false);
+    const [isSwiping, setIsSwiping] = useState(false);
+
+    const animationTimerRef = useRef<ReturnType<typeof setTimeout>>();
+    const swipingTimerRef = useRef<ReturnType<typeof setTimeout>>();
+    const scrollableContainerRef = useRef<HTMLDivElement>(null);
+
+    const modalRoot =
+        (portalId && document.getElementById(portalId)) ||
+        document.getElementById("modal-root") ||
+        document.body;
+
+    const shouldExpand =
```

Comment:
If content is too long, then modal should open on 90% of height. In this case it doesn't make sense to collapse it on swipe down, so, as discussed with designers, it should be closed on swipe down.
This flag is needed to track if modal takes more then 85% of height without expanding